x-pack/plugins/enterprise_search/public/applications/app_search/components/documents/document_detail_logic.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+import { i18n } from '@kbn/i18n';
+
+import { HttpLogic } from '../../../shared/http';
+import { EngineLogic } from '../engine';
+import { flashAPIErrors, setQueuedSuccessMessage } from '../../../shared/flash_messages';
+import { FieldDetails } from './types';
+
+export interface DocumentDetailLogicValues {
+  dataLoading: boolean;
+  fields: FieldDetails[];
+}
+
+export interface DocumentDetailLogicActions {
+  setFields(fields: FieldDetails[]): { fields: FieldDetails[] };
+  deleteDocument(documentId: string): { documentId: string };
+  getDocumentDetails(documentId: string): { documentId: string };
+}
+
+type DocumentDetailLogicType = MakeLogicType<DocumentDetailLogicValues, DocumentDetailLogicActions>;
+
+const CONFIRM_DELETE = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.documentDetail.confirmDelete',
+  {
+    defaultMessage: 'Are you sure you want to delete this document?',
+  }
+);
+const DELETE_SUCCESS = i18n.translate(
```

Comment:
[Not blocking] No strong preference either way, but any thoughts on moving copy out to a `constants.ts` file? 🤷

Reply:
I know we've done that elsewhere, and I did consider it. I guess in this particular case I didn't have a strong reason to move them out so I just left them in here.

Reply:
I have a hunch over time we'll find it easier to quickly read logic files the less there is at the top of them (e.g. imports -> type defs -> straight to logic def), but definitely not a blocker as I mentioned